src/gradlecheck/OpenSearchMetricsQuery.groovy

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package gradlecheck
+
+import groovy.json.JsonSlurper
+
+class OpenSearchMetricsQuery {
+    String metricsUrl
+    String awsAccessKey
+    String awsSecretKey
+    String awsSessionToken
+    def script
+
+    OpenSearchMetricsQuery(String metricsUrl, String awsAccessKey, String awsSecretKey, String awsSessionToken, def script) {
+        this.metricsUrl = metricsUrl
+        this.awsAccessKey = awsAccessKey
+        this.awsSecretKey = awsSecretKey
+        this.awsSessionToken = awsSessionToken
+        this.script = script
+    }
+
+    def fetchMetrics(String query) {
+        def response = script.sh(
```

Comment:
just for my knowledge, what is 'script.sh'?

Reply:
The `script` is a jenkins context and will be injected to the library during runtime. Without the jenkins context the native groovy does have the `sh` inbuit.

Reply:
Adding more to this, to run a script in groovy the ideal way to use is `.execute()`, but running `.execute()` in jenkins is blocked by the script security plugin, hence the ideal was is to use `sh` given by jenkins and for `sh` to work in libraries it needs jenkins context.